src/main/java/com/redhat/devtools/lsp4ij/features/foldingRange/LSPFoldingRangeBuilder.java

```diff
@@ -54,8 +66,30 @@ protected void buildLanguageFoldRegions(@NotNull List<FoldingDescriptor> descrip
         }
 
         PsiFile file = root.getContainingFile();
+        List<FoldingRange> foldingRanges = getFoldingRanges(file);
+        if (!ContainerUtil.isEmpty(foldingRanges)) {
+            for (FoldingRange foldingRange : foldingRanges) {
+                TextRange textRange = getTextRange(foldingRange, document);

```

Comment:
And again, adapt each folding range to the corresponding text range. The only reason a null text range would be returned is of the language server returned a folding range that didn't make sense, e.g., end before start.